sklearn/ensemble/tests/test_forest.py

```diff
@@ -371,6 +376,86 @@ def test_unfitted_feature_importances(name):
         getattr(FOREST_ESTIMATORS[name](), 'feature_importances_')
 
 
+@pytest.mark.parametrize("name", FOREST_CLASSIFIERS_REGRESSORS)
+@pytest.mark.parametrize("dtype", (np.float64, np.float32))
+def test_oob_importances(name, dtype):
+
+    X = X_large.astype(dtype=dtype, copy=False)
+    y = y_large.astype(dtype=dtype, copy=False)
+
+    ForestEstimator = FOREST_ESTIMATORS[name]
+
+    clf = ForestEstimator(
+        n_estimators=10,
+        random_state=0,
+        feature_importances_type="permutation",
+        bootstrap=True,
+    )
+    clf.fit(X, y)
+    importances = clf.feature_importances_
+    if name in FOREST_CLASSIFIERS:
+        imp_level = 0.025
+    else:
+        imp_level = 0.1
+
+    n_important = np.sum(importances > imp_level)
+    assert importances.shape[0] == 10
+    assert n_important == 3
+    assert np.all(importances[:3] > imp_level)
+
+
+@pytest.mark.parametrize("name", FOREST_CLASSIFIERS_REGRESSORS)
+@pytest.mark.parametrize("dtype", (np.float64, np.float32))
+def test_oob_importances_multi_class(name, dtype):
+
+    X = X_large_multiclass.astype(dtype=dtype, copy=False)
+    y = y_large_multiclass.astype(dtype=dtype, copy=False)
+
+    ForestEstimator = FOREST_ESTIMATORS[name]
+
+    clf = ForestEstimator(
+        n_estimators=10,
+        random_state=0,
+        feature_importances_type="permutation",
+        bootstrap=True,
+    )
+    clf.fit(X, y)
+    importances = clf.feature_importances_
+
+    n_important = np.sum(importances > 0.05)
+    assert importances.shape[0] == 10
+    assert n_important == 3
+    assert np.all(importances[:3] > 0.05)
+
+
+@pytest.mark.parametrize("name", FOREST_ESTIMATORS)
+def test_oob_importances_raise_error(name):
+    ForestEstimator = FOREST_ESTIMATORS[name]
+
+    if name in FOREST_TRANSFORMERS:
```

Comment:
Here, you should be able to parametrize the function regarding the type of error raised, the error message and the parameter passed to the estimator and just call `estimator.fit(X, y)` to check that the error is raised.

Use pytest to check the error message

```python
with pytest.raises(ValueError, match="partial error message"):
    call_raising_error(....)
```